v3.4.x Typescript issue in RetryConfig in index.d.ts

#### What are you trying to achieve?
Use retry as RetryConfig or RetryConfig[] without defining all fields.
So I can't use retry as indicated in examples here https://codecept.io/basics/#retry-configuration

#### What do you get instead?
Typescript errors

```bash
TSError: ⨯ Unable to compile TypeScript:
codecept.conf.ts:19:3 - error TS2322: Type '{ Feature: number; Scenario: number; Before: number; BeforeSuite: number; After: number; AfterSuite: number; }' is not assignable to type 'number | RetryConfig | RetryConfig[]'.
  Type '{ Feature: number; Scenario: number; Before: number; BeforeSuite: number; After: number; AfterSuite: number; }' is not assignable to type 'number'.
```

> Provide test source code if related
This doesn't work
```js
  retry: {
    Scenario: 3,
    Before: 3,
  },
```
And this as well
```js
  retry: [{
    Scenario: 3,
    Before: 3,
  }],
```

I can solve the issue by specifying all fields like:
```js
  retry: {
    grep: '',
    Feature: 0,
    Scenario: 3,
    Before: 3,
    BeforeSuite: 3,
    After: 0,
    AfterSuite: 0,
  },
```

### Details

* CodeceptJS version: 3.4.1


```js
import { setHeadlessWhen, setCommonPlugins } from '@codeceptjs/configure';
// turn on headless mode when running with HEADLESS=true environment variable
// export HEADLESS=true && npx codeceptjs run
setHeadlessWhen(process.env.HEADLESS);

// enable all common plugins https://github.com/codeceptjs/configure#setcommonplugins
setCommonPlugins();

export const config: CodeceptJS.MainConfig = {
  tests: './*_test.ts',
  output: './output',
  helpers: {
    Playwright: {
      url: 'https://codecept.io/',
      show: false,
      browser: 'chromium'
    }
  },
  retry: {
    //grep: '',
    Feature: 0,
    Scenario: 3,
    Before: 3,
    BeforeSuite: 3,
    After: 0,
    AfterSuite: 0,
  },
  include: {
    I: './steps_file'
  },
  name: 'CodeceptJSExample'
}
```

You can generate the issue here https://github.com/MhdAlkh/BeforeHookIssue

--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -19,19 +19,19 @@ declare namespace CodeceptJS {
 
   type RetryConfig = {
     /** Filter tests by string or regexp pattern */
-    grep: string | RegExp;
+    grep?: string | RegExp;
     /** Number of times to repeat scenarios of a Feature */
-    Feature: number;
+    Feature?: number;
     /** Number of times to repeat scenarios */
-    Scenario: number;
+    Scenario?: number;
     /** Number of times to repeat Before hook */
-    Before: number;
+    Before?: number;
     /** Number of times to repeat After hook */
-    After: number;
+    After?: number;
     /** Number of times to repeat BeforeSuite hook */
-    BeforeSuite: number;
+    BeforeSuite?: number;
     /** Number of times to repeat AfterSuite hook */
-    AfterSuite: number;
+    AfterSuite?: number;
   };
 
   type TimeoutConfig = {
@@ -392,7 +392,7 @@ declare namespace CodeceptJS {
     | { ios: string }
     | { android: string; ios: string }
     | { react: string }
-    | { shadow: string }
+    | { shadow: string[] }
     | { custom: string };
 
   interface CustomLocators {}

